src/container/srv_target.c

```diff
@@ -99,18 +102,15 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *sleep)
 	if (rc)
 		return rc;
 
-	/*
-	 * sc_aggregation_min != DAOS_EPOCH_MAX means we need to aggregate
-	 * from epoch 0 because some snapshot was deleted.
-	 */
-	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX) {
-		epoch_min = 0;
-		/*
-		 * Set sc_aggregation_min to non-zero so that we can tell
-		 * if another snapshot deletion happened when this round of
-		 * aggregation done.
+	change_hlc = max(cont->sc_snapshot_delete_hlc,
+			 cont->sc_pool->spc_rebuild_end_hlc);
```

Comment:
seems in some cases, it is possible the spc_rebuild_fence > spc_rebuild_end_hlc

Reply:
yes, that is possible, spc_rebuild_end_hlc basically means the last ended rebuild hlc. but spc_rebuild_fence is the current ongoing rebuild.  end_hlc is used to compare with the aggregation start hlc to know whether it needs to restart aggregation from 0, while rebuild_fence is used to indicate aggregate to skip this rebuild fence, instead of aggregating it, to avoid interfering the rebuild.  Or I miss your question?